src/Site.js

```diff
@@ -894,6 +897,26 @@ Site.prototype.copyFontAwesomeAsset = function () {
   return fs.copyAsync(faCssSrcPath, faCssDestPath).then(fs.copyAsync(faFontsSrcPath, faFontsDestPath));
 };
 
+/**
+ * Minifies and copies Octicon assets to the assets folder
+ */
+Site.prototype.copyOcticonsAsset = function () {
+  const octiconsRootSrcPath = path.join(__dirname, '..', 'node_modules', '@primer', 'octicons', 'build');
+  const octiconsCssSrcPath = path.join(octiconsRootSrcPath, 'build.css');
+  const octiconsCssDestPath = path.join(this.siteAssetsDestPath, 'css', 'octicons.min.css');
+
+  const minifiedCss = fs.readFileSync(octiconsCssSrcPath).toString().replace(/\s/g, '');
```

Comment:
Sorry I missed this out previously.

I don't think we should be manually minifying CSS like this. It might be fine _for now_, but it might introduce some very hard to find errors if any whitespace sensitive CSS properties are ever used in the Octicon CSS file. 

If there is no minified CSS file included in the library, I think we can just use the un-minified CSS for now since it's not very big. Perhaps we should look into programatically minifying all of our CSS and JS assets during the build phase.

Reply:
That's actually the approach I initially went with. The problem is - The unminified CSS uses 2 spaces for indentation, which causes stylelint violations (expects 4 spaces) when the test sites are updated with the Octicon CSS file (via `npm run updatetest`). 

I guess here we have a few options: 

1. If we're worried about self-minification being fragile, we can consider bringing in a library to help with this - [`clean-css`](https://github.com/jakubpawlowicz/clean-css) for example. 

2. Add rules for stylelint to ignore the functional test sites.

Reply:
Ah that's unfortunate. Applying our own linting rules to external dependencies will probably always be an exercise in frustration. I'm leaning towards making stylelint ignore the functional test sites. 

@yamgent do you have any thoughts on this?

Reply:
@marvinchin how should we proceed from here?

Reply:
Let's go ahead and disable the lint rules for the test sites (lets try to make the excluded paths as specific as possible). We should do that in a separate PR, and then rebase this PR on top of that.

Reply:
@marvinchin I was thinking, we could simply name the Octicon CSS file `octicons.min.css` when copying without actually minifying it. 

This allows stylelint to ignore it (because it ends with `min.css`, just like all our other external libraries) without having to modify the project configuration just for one rare case of a library that doesn't distribute minified files. It also allows us to continue linting the test sites without having to add library-specific ignore rules. What do you think?

Reply:
@dvrylc I’d prefer not to do that because the CSS file is not actually minified, so naming it as such would be confusing. (It would also be achieving the same outcome of “disabling” the linter for the file, but in a more hacky way)

Not linting generated files seem to make sense to me because some of these generated files might be minified, dependent on external assets etc. and would not satisfy our linting rules (it wouldn’t make sense to lint node_modules for similar reasons!). We should only be applying linting rules to the source files that we are working on.

Also, to clarify - I’m suggesting excluding all generated files from the linter, and not specifically this file in particular.

Happy to discuss this further if you have any other thoughts or concerns!

Reply:
Ah I see your point, sounds good! I'll rebase and update this once #1011 is merged.